### PR TITLE
added scanreport and dataset permission classes

### PR DESCRIPTION
**Pull request by Samad198**

### CODE REVIEW

File: api/mapping/permissions.py [hunk 1]
```diff
@@ -9,3 +12,58 @@ def has_object_permission(self, request, view, obj):
...
+
+
+class CanViewDataset(permissions.BasePermission):
+
+    message = "You do not have permission to view this dataset"
+
+    def has_object_permission(self, request, view, obj):
+        """
+        Return `True` if the User's ID is in the Project's members.
+        """
```
Comment (dcl10) [inline api/mapping/permissions.py:24]:
```suggestion
        """
        Return `True` if, the dataset is 'public' and the User's ID is in the Project's members, or, the dataset is 'restricted' and the User's ID is in a project the User is a member of. 
        """
```

File: api/mapping/permissions.py [hunk 2]
```diff
@@ -9,3 +12,58 @@ def has_object_permission(self, request, view, obj):
...
+        # check if the user is in the viewers field
+        if visibility == "RESTRICTED":
+            self.message = "You must be granted permission to view this dataset"
+            return obj.viewers.filter(id=request.user.id).exists()
+        # if the visibility is public
+        # check if the user is a member in any projects the parent dataset is in
+        elif visibility == "PUBLIC":
+            # filter by projects that have dataset obj.parent_dataset
+            # filter by projects that have user as a member
+            self.message = "You are not part of any projects for this dataset"
```
Comment (dcl10) [inline api/mapping/permissions.py:37]:
```suggestion
            self.message = "You are not a member of any projects for this dataset"
```

File: api/mapping/permissions.py [hunk 3]
```diff
@@ -9,3 +12,58 @@ def has_object_permission(self, request, view, obj):
...
+            return Project.objects.filter(
+                datasets__id=obj.id, members__id=request.user.id
+            ).exists()
+        print("VISIBILITY IS", visibility)
```
Comment (dcl10) [inline api/mapping/permissions.py:41]:
```suggestion
```

File: api/mapping/permissions.py [hunk 4]
```diff
@@ -9,3 +12,58 @@ def has_object_permission(self, request, view, obj):
...
+        if visibility == "RESTRICTED":
+            self.message = "You must be granted permission to view this scan report"
+            return obj.viewers.filter(id=request.user.id).exists()
+        # if the visibility is public
+        # check if the user is a member in any projects the parent dataset is in
+        elif visibility == "PUBLIC":
+            # get projects
+            # filter by projects that have dataset obj.parent_dataset
+            # filter by projects that have user as a member
+            self.message = "You are not part of any projects for this scan report"
```
Comment (dcl10) [inline api/mapping/permissions.py:64]:
```suggestion
            self.message = "You are not a member of any projects for this scan report"
```

File: api/mapping/permissions.py [hunk 5]
```diff
@@ -9,3 +12,58 @@ def has_object_permission(self, request, view, obj):
...
+        print("VISIBILITY IS", visibility)
+        return False
+
+
+class CanViewScanReport(permissions.BasePermission):
+    message = "You do not have permission to view this"
+
+    def has_object_permission(self, request, view, obj):
+        """
+        Return `True` if the User's ID is in the Project's members.
```
Comment (dcl10) [inline api/mapping/permissions.py:50]:
```suggestion
        Return `True` if, the scan report is 'public' and the User's ID is in the Project's members, or, the scan report is 'restricted' and the User's ID is in a project the User is a member of.
```

**Review by dcl10 — CHANGES_REQUESTED** [review]

**Comment by Samad198** [comment]
Public
image

Restricted 
image

**Review by dcl10 — APPROVED** [review]
LGTM

**Comment by Samad198** [comment]
image


image

**Review by dcl10 — APPROVED** [review]

File: api/mapping/urls.py [hunk 6]
```diff
@@ -150,6 +149,16 @@
         views.DatasetRetrieveView.as_view(),
         name="datasets_retrieve",
     ),
+    # path(
+    #     r"api/scanreports/",
+    #     views.ScanReportListViewSet.as_view(),
+    #     name="scanreport_list",
+    # ),
```
Comment (calmacx) [inline api/mapping/urls.py:156]: removed commented out code

**Review by calmacx — APPROVED** [review]
approved if you want to make that small change to remove commented out code and then merge

**Pull request merged**